carp.deployments.core/src/commonTest/kotlin/dk/cachet/carp/deployments/application/DeploymentServiceTest.kt

```diff
@@ -178,6 +178,28 @@ interface DeploymentServiceTest
         assertTrue( device in status.getRemainingDevicesToRegister() )
     }
 
+    @Test
+    fun getDeviceDeploymentFor_during_device_reregistrations() = runTest {
+        val (service, _) = createService()
+        val protocol = createSinglePrimaryDeviceProtocol( "Test device" )
+        val device = protocol.primaryDevices.first()
+        val deploymentId = UUID.randomUUID()
+        service.createStudyDeployment( deploymentId, protocol.getSnapshot(), listOf( createParticipantInvitation() ) )
+        val firstRegistration = device.createRegistration()
+        service.registerDevice( deploymentId, device.roleName, firstRegistration )
+
+        val firstDeployment: PrimaryDeviceDeployment = service.getDeviceDeploymentFor( deploymentId, device.roleName )
+        assertEquals( firstRegistration, firstDeployment.registration )
+
+        service.unregisterDevice( deploymentId, device.roleName )
+        assertFailsWith<IllegalArgumentException> { service.getDeviceDeploymentFor( deploymentId, device.roleName ) }
```

Comment:
Could you explain why do you throw an exception when there is no `DeviceDeploymentFor( deploymentId, device.roleName )`. My intuition is returning something likes empty/null would make more sense. Unless this exception will be interpreted by backend implementation.

Reply:
Oh, the document says  `@throws IllegalArgumentException if [dataStream] has never been opened`. But in this case it has been opened before, doesn't it?

Reply:
Data streams are totally unrelated here. I think you got confused.

But to clarify your original question, there seems to be documentation missing on that call which I'm adding now.

> @throws IllegalArgumentException when:
>   - the device with [primaryDeviceRoleName] has not yet been registered

Reply:
That makes sense.